--- src/config.c~	2013-02-09 11:17:04.299988149 +0800
+++ src/config.c	2013-02-09 11:17:51.869987806 +0800
@@ -676,6 +676,7 @@
 		}
 	}
 	else if (strcmp(key, "battery") == 0) {
+#ifdef ENABLE_BATTERY
 		if (new_config_file == 0) {
 			battery_enabled = atoi(value);
 			if (battery_enabled) {
@@ -688,6 +689,7 @@
 					panel_items_order = g_strdup("B");
 			}
 		}
+#endif
 	}
 	else
 		fprintf(stderr, "tint2 : invalid option \"%s\",\n  upgrade tint2 or correct your config file\n", key);
