--- src/xfns.orig.c	2013-04-18 10:00:35.259942096 +0800
+++ src/xfns.c	2013-04-18 10:01:19.219941779 +0800
@@ -2190,6 +2190,10 @@
 	  supported_list.supported_styles = supported_xim_styles;
 	  xic_style = best_xim_style (&supported_list,
 				      FRAME_X_XIM_STYLES (f));
+          if (!xfs) {
+            printf("Unable to create fontset for XIM, disabling Preedit.\n");
+            xic_style &= ~(XIMPreeditPosition | XIMPreeditArea);
+          }
 	}
 
       preedit_attr = XVaCreateNestedList (0,
@@ -2221,6 +2225,9 @@
 		       XNStatusAttributes, status_attr,
 		       XNPreeditAttributes, preedit_attr,
 		       NULL);
+      if (!xic) {
+        printf("Failed to create XIM input context with style %#06lx, window %#010lx\n", xic_style, FRAME_X_WINDOW(f));
+      }
       XFree (preedit_attr);
       XFree (status_attr);
     }
