--- xfwm4-4.10.1/src/tabwin.c~	2013-07-25 20:45:27.009966473 +0800
+++ xfwm4-4.10.1/src/tabwin.c	2013-07-25 20:45:34.109966422 +0800
@@ -380,6 +380,7 @@
     tbw->selected_callback = 0;
 
     gtk_window_set_screen (GTK_WINDOW (tbw), screen_info->gscr);
+    gtk_window_set_title (GTK_WINDOW (tbw), "xfwm4-tabwin");
     gtk_widget_set_name (GTK_WIDGET (tbw), "xfwm4-tabwin");
     gtk_widget_realize (GTK_WIDGET (tbw));
     gtk_container_set_border_width (GTK_CONTAINER (tbw), 0);
