--- wine-1.9.9/dlls/gdi32/freetype.c~	2016-04-30 00:04:00.000000000 +0800
+++ wine-1.9.9/dlls/gdi32/freetype.c	2016-05-15 11:42:17.320142986 +0800
@@ -4025,8 +4025,6 @@
 
             if (!RegCreateKeyA( HKEY_LOCAL_MACHINE, "Software\\Microsoft\\Windows NT\\CurrentVersion\\FontSubstitutes", &hkey ))
             {
-                RegSetValueExA(hkey, "MS Shell Dlg", 0, REG_SZ, (const BYTE *)nls_update_font_list[i].shelldlg,
-                               strlen(nls_update_font_list[i].shelldlg)+1);
                 RegSetValueExA(hkey, "Tms Rmn", 0, REG_SZ, (const BYTE *)nls_update_font_list[i].tmsrmn,
                                strlen(nls_update_font_list[i].tmsrmn)+1);
 
