--- wine-1.7.0/dlls/gdi32/freetype.c~	2013-08-03 02:09:56.000000000 +0800
+++ wine-1.7.0/dlls/gdi32/freetype.c	2013-09-20 20:24:25.770082226 +0800
@@ -3833,8 +3833,6 @@
 
             if (!RegCreateKeyA( HKEY_LOCAL_MACHINE, "Software\\Microsoft\\Windows NT\\CurrentVersion\\FontSubstitutes", &hkey ))
             {
-                RegSetValueExA(hkey, "MS Shell Dlg", 0, REG_SZ, (const BYTE *)nls_update_font_list[i].shelldlg,
-                               strlen(nls_update_font_list[i].shelldlg)+1);
                 RegSetValueExA(hkey, "Tms Rmn", 0, REG_SZ, (const BYTE *)nls_update_font_list[i].tmsrmn,
                                strlen(nls_update_font_list[i].tmsrmn)+1);
 
