diff --git a/gui/vertexdatainterpreter.cpp b/gui/vertexdatainterpreter.cpp
index a301231..ae12678 100644
--- a/gui/vertexdatainterpreter.cpp
+++ b/gui/vertexdatainterpreter.cpp
@@ -192,7 +192,7 @@ void VertexDataInterpreter::interpretData()
     default:
         qDebug()<<"unkown gltype = "<<m_type;
     }
-    //qDebug()<<"list is "<<lst;
+    qDebug() << "Interpreted vertex data: " << lst;
     m_listWidget->addItems(lst);
 }
 
